### DISCUSSION TITLE
On the usage of 'line gap width'

### DISCUSSION THREAD

**Discussion by mdxiaohu** [maplibre/maplibre-gl-js#2800] [Q&A]
## User Story

I use the 'line-gap-width' attribute .
The assignment is as follows .

```
"line-width": 4,
line-gap-width: 2
```

Page Effects .

![Snipaste_2023-07-04_11-48-35](https://github.com/maplibre/maplibre-gl-js/assets/42248614/ab825afe-a97c-4d98-be44-921a09888ff8)

## Rationale

I'm not sure if it's possible to set this attribute to achieve the effect shown in the following image .

![Snipaste_2023-07-04_11-57-01](https://github.com/maplibre/maplibre-gl-js/assets/42248614/4ad40afc-51ea-47c8-897f-4d24596f4cee)

As shown in the figure, there are gaps in the lines .
The difference is that .
The virtual line can be set to its width .
Instead of using the width of the original line .

## Impact

I am allowed to use icons to complete this function, but I am not sure if there are any other feasible methods .
For example, modifying the width of virtual lines .

**Comment by HarelM** [comment]
Is this an issue it a question?
You can use different line layers with different widths using offset I believe.

**Comment by mdxiaohu** [comment]
Perhaps my description is not clear enough .
I use a geojson data source .
Created a layer .
For example, the following example .

`https://jsfiddle.net/mdxiaohu/4j9b5m8d/3/`

I don't know if a single layer can achieve this effect .
I saw the "line-gap-width" attribute and created a virtual line .
So, I was wondering if it's possible to modify the width of this virtual line since it can be created .
Perhaps this method is not feasible .
If a single layer is not possible, I will consider using image filling to achieve a double line style .

**Comment by HarelM** [comment]
I have no clue what is a virtual line.
As I said, you can create two lines, each with a different width.
